lib/trento/infrastructure/alerting/alerting.ex

```diff
@@ -74,6 +79,34 @@ defmodule Trento.Infrastructure.Alerting.Alerting do
     )
   end
 
+  defp maybe_notify_api_key_expiration(false), do: :ok
+
+  defp maybe_notify_api_key_expiration(true) do
+    case Settings.get_api_key_settings() do
+      {:ok, %ApiKeySettings{expire_at: nil}} ->
+        :ok
+
+      {:ok, %ApiKeySettings{} = api_key_settings} ->
+        maybe_send_api_key_notification(api_key_expiration_days(api_key_settings))
+
+      error ->
+        error
+    end
+  end
+
+  defp maybe_send_api_key_notification(days) when days < 0 do
+    deliver_notification(EmailAlert.api_key_expired())
+  end
+
+  defp maybe_send_api_key_notification(days) when days < 30 do
+    deliver_notification(EmailAlert.api_key_will_expire(days))
+  end
+
+  defp maybe_send_api_key_notification(_), do: :ok
+
+  defp api_key_expiration_days(%ApiKeySettings{expire_at: expire_at}),
```

Comment:
This should be moved above `maybe_send_api_key_notification` as it is called before

Reply:
done